bench-vortex/src/statpopgen/download_vcf.rs

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright the Vortex contributors
+
+use std::io;
+use std::sync::Arc;
+
+use futures::StreamExt;
+use indicatif::ProgressBar;
+use noodles_vcf::Record;
+use parquet::arrow::{AsyncArrowWriter, ParquetRecordBatchStreamBuilder};
+use reqwest::Client;
+use tokio::fs::{File, create_dir_all};
+use tokio::io::BufReader;
+use tokio::runtime::Handle;
+use tokio_util::io::StreamReader;
+use tracing::info;
+use vortex::ArrayRef;
+use vortex::arrow::FromArrowArray;
+use vortex::compressor::CompactCompressor;
+use vortex::dtype::DType;
+use vortex::dtype::arrow::FromArrowType;
+use vortex::error::{VortexError, VortexResult, vortex_bail, vortex_err};
+use vortex::file::{VortexWriteOptions, WriteStrategyBuilder};
+use vortex::stream::ArrayStreamAdapter;
+
+use super::StatPopGenBenchmark;
+use crate::statpopgen::builder::GnomADBuilder;
+use crate::statpopgen::schema::SCHEMA;
+use crate::{Format, idempotent_async};
+
+// DuckDB parallelizes parquet at row-group granularity. Each of our rows are quite big (~4000
+// genotypes each with tens of bytes of data).
+const ROW_GROUP_SIZE_IN_VARIANTS: u64 = 1024;
+
+impl StatPopGenBenchmark {
+    pub async fn download_parquet(&self) -> VortexResult<()> {
+        let url = "https://gnomad-public-us-east-1.s3.amazonaws.com/release/3.1.2/vcf/genomes/gnomad.genomes.v3.1.2.hgdp_tgp.chr21.vcf.bgz";
```

Comment:
I assume the dataset isn't available directly somewhere as parquet?

Reply:
It is not. It is available as Hail!

Reply:
FWIW, BGZIP-compressed, plaintext VCFs are the defacto interchange standard of genetics.